int32 x
int32 y
int32 leftupper_x
int32 leftupper_y
int32 rightlower_w
int32 rightlower_h
